### ISSUE TITLE
Missing functionalities

### ISSUE THREAD

**Issue by AKJ7** [riptideio/pymodbus#1208]
### Versions

* Python: 3.10
* OS: Linux
* Pymodbus: Latest
* Modbus Hardware (if used):

### Pymodbus Specific
* Server: TCP- Async
* Client: TCP - Async

### Description
When creating an async TCP Client, the following functionalities, despite been valid are missing:
- `timeout`
- retires
- close_comm_on_error

Moreover, on-connection-error (see [me](https://github.com/riptideio/pymodbus/blob/v2.5.3/pymodbus/client/asynchronous/async_io/__init__.py#L624)), pyModbus doesn't provide any mechanism to wait or cancel the connection-retry (the solution is simple. Just return the Coroutines).

### Code and Logs

```python
from pymodbus.client.tcp import AsyncModbusTcpClient as ModbusClient

client = ModbusClient('localhost', retries=10, timeout=3, close_comm_on_error=True)
connect = await client.connect()
```

**Comment by janiversen** [comment]
Your example tells me you are not using the latest version (3.0.2) but using version 2.5.3.

timeout is implemented. But you are right about retries and close_comm_on_error.

Regarding the connect() you can use wait_for() to stop the initial connect(). To terminate the retries, simply delete the client object. 

I have no idea what you mean by "just return the Coroutines"

Pull requests are more than welcome.

**Comment by janiversen** [comment]
No, I am not sure you can do that at all with python. The modbus protocol does not know about interrupts.

**Comment by janiversen** [comment]
Let’s keep it open until the codebase is updated.
